tests/test_sample_projects.py

```diff
@@ -20,65 +19,94 @@
 
 import pytest
 
 from fawltydeps.main import Analysis
 from fawltydeps.settings import Action, Settings
+from fawltydeps.types import TomlData
 
-if sys.version_info >= (3, 11):
-    import tomllib  # pylint: disable=E1101
-else:
-    import tomli as tomllib
+from .project_helpers import BaseExperiment, BaseProject, parse_toml
 
 # These are (slow) integration tests that are disabled by default.
 pytestmark = pytest.mark.integration
 
 SAMPLE_PROJECTS_DIR = Path(__file__).with_name("sample_projects")
 
-sample_projects_params = [
-    pytest.param(sample_project, id=sample_project.name)
-    for sample_project in SAMPLE_PROJECTS_DIR.iterdir()
-    if sample_project.is_dir()
-]
-
-
-@pytest.mark.parametrize("project_path", sample_projects_params)
-def test_integration_analysis_on_sample_projects__(project_path):
-    settings = Settings(
-        actions={Action.REPORT_UNDECLARED, Action.REPORT_UNUSED},
-        code=[project_path],
-        deps=[project_path],
-    )
-    analysis = Analysis.create(settings)
-
-    with (project_path / "expected.toml").open("rb") as f:
-        experiment = tomllib.load(f)
-
-    print(experiment["project"].get("name"))
-    print(experiment["project"].get("description"))
-    expected = experiment.get("analysis_result", {})
-
-    if "imports" in expected:
-        print("Verifying 'imports'...")
-        actual_imports = {i.name for i in analysis.imports}
-        expect_imports = set(expected["imports"])
-        assert actual_imports == expect_imports
-
-    if "declared_deps" in expected:
-        print("Verifying 'declared_deps'...")
-        actual_declared = {d.name for d in analysis.declared_deps}
-        expect_declared = set(expected["declared_deps"])
-        assert actual_declared == expect_declared
-
-    print("Verifying 'undeclared_deps'...")
-    actual_undeclared = {u.name for u in analysis.undeclared_deps}
-    expect_undeclared = set(expected.get("undeclared_deps", []))
-    assert actual_undeclared == expect_undeclared
-
-    print("Verifying 'unused_deps'...")
-    actual_unused = {u.name for u in analysis.unused_deps}
-    expect_unused = set(expected.get("unused_deps", []))
-    assert actual_unused == expect_unused
+
+@dataclass
+class Experiment(BaseExperiment):
+    """A single experiment to run FawltyDeps on a sample project.
+
+    Input to the experiment consists of the following members:
+    - code: Settings.code paths relative to the sample project root
+    - deps: Settings.deps paths relative to the sample project root
+
+    See BaseExperiment for details on the inherited members.
+    """
+
+    code: List[str]
+    deps: List[str]
+
+    @classmethod
+    def from_toml(cls, name: str, data: TomlData) -> "Experiment":
+        return cls(
+            code=data.get("code", [""]),
+            deps=data.get("deps", [""]),
+            **cls._init_args_from_toml(name, data),
+        )
+
+    def settings(self, project_path: Path, cache: pytest.Cache) -> Settings:
```

Comment:
Nit: slight preference with this named with a prefix that's more indicative of the action, e.g. `as_settings`, `to_settings`, `build_settings`, `create_settings`

Reply:
Probably `build_settings` or `create_settings` among those 4 mentioned above, just since addition of additional data is needed, rather than simply converting the existing instance